lib/ansible/modules/cloud/vmware/vmware_vsan_health_facts.py

```diff
@@ -0,0 +1,170 @@
+- name: Gather health facts from a vSAN's cluster
+  hosts: localhost
+  gather_facts: 'no'
+  vmware_vsan_health_facts:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    cluster_name: 'vSAN01'
+    fetch_from_cache: 'False'
+  register: facts
+'''
+
+RETURN = '''
+health:
+    description: vSAN cluster health facts
+    type: dict
+    sample: {
+        "_vimtype": "vim.cluster.VsanClusterHealthSummary",
+        "burnInTest": null,
+        "clusterStatus": {
+            "_vimtype": "vim.cluster.VsanClusterHealthSystemStatusResult",
+            "goalState": "installed",
+            "status": "green",
+            "trackedHostsStatus": [
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi01.example.com",
+                    "issues": [],
+                    "status": "green"
+                },
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi04.example.com",
+                    "issues": [],
+                    "status": "green"
+                },
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi02.example.com",
+                    "issues": [],
+                    "status": "green"
+                },
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi03.example.com",
+                    "issues": [],
+                    "status": "green"
+                }
+            ],
+            "untrackedHosts": []
+        }
+    }
+'''
+
+try:
+    from pyVmomi import vim, vmodl, VmomiSupport
+    HAS_PYVMOMI = True
+    HAS_PYVMOMIJSON = hasattr(VmomiSupport, 'VmomiJSONEncoder')
+except ImportError:
+    HAS_PYVMOMI = False
+    HAS_PYVMOMIJSON = False
+
+try:
+    import vsanapiutils
+    import vsanmgmtObjects
+    HAS_VSANPYTHONSDK = True
+except ImportError:
+    HAS_VSANPYTHONSDK = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import (connect_to_api, vmware_argument_spec, find_cluster_by_name)
+
+import json
+
+
+def run_module():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(dict(cluster_name=dict(required=True, type='str')))
+    argument_spec.update(dict(fetch_from_cache=dict(required=False, type='bool')))
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
+
+    if not HAS_PYVMOMI:
+        module.fail_json(msg='pyvmomi is required for this module')
+
+    if not HAS_VSANPYTHONSDK:
+        module.fail_json(msg='vSAN python SDK is required for this module')
+
+    if not HAS_PYVMOMIJSON:
+        self.module.fail_json(msg='The installed version of pyvmomi lacks JSON output support; need pyvmomi>6.7.1')
```

Comment:
I think `self. ` should be removed here
```suggestion
        module.fail_json(msg='The installed version of pyvmomi lacks JSON output support; need pyvmomi>6.7.1')
```

Reply:
Good catch